desc/efit.py

```diff
@@ -0,0 +1,374 @@
+"""Class for reading and processing EFIT files."""
+
+import numpy as np
+from matplotlib import pyplot as plt
+from scipy.interpolate import CubicSpline
+
+from desc.basis import DoubleFourierSeries
+from desc.equilibrium import Equilibrium
+from desc.geometry import FourierRZToroidalSurface
+from desc.grid import LinearGrid
+from desc.profiles import PowerSeriesProfile, SplineProfile
+from desc.transform import Transform
+
+
+class EFITIO:
+    """Performs input from EFIT files to DESC Equilibrium."""
+
+    @classmethod
+    def load(  # noqa: C901 - fxn too complex
+        cls,
+        g,
+        M,
+        N=0,
+        L=None,
+        bdry_dist=1,
+    ):
+        """Create a DESC equilibrium object from an EFIT geqdsk file.
+
+        The equilibrium will not be solved, this just creates an equilibirum
+        with the correct boundary and profiles.
+
+        Parameters
+        ----------
+        g : path-like or dict
+            Path to geqdsk file, or dict of already read parameters.
+        M : int
+            Poloidal resolution of the equilibrium. Used for fitting the boundary.
+        N : int, optional
+            Toroidal resolution of the equilibrium.
+        L : int, optional
+            Radial resolution of the equilibrium. Used for fitting profiles.
+            If None, spline profiles will be used. Otherwise they are fit
+            to even power series in rho.
+        bdry_dist : float
+            Deviation from separatrix. If bdry_dist == 1, then the boundary
```

Comment:
is this related to like $\rho$ or $\psi_n$?

Reply:
It's normalized poloidal flux. Adding that to the docstring.